### PR TITLE
Add CancellationToken parameters

### PR DESCRIPTION
**Pull request by Reminouche**
I added a CancellationToken paramter to InvokeApi functions.
This modification was requested here: https://github.com/Azure/azure-mobile-services/issues/524

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@Reminouche__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@Reminouche__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

**Comment by paulbatum** [comment]
We'd like to see this change go in, but unfortunately this pull request adds duplicate implementations of RequestAsync, SendRequestAsync and InvokeApiAsync. Any chance you can refactor the existing method implementations to avoid the copy/paste?

**Comment by Reminouche** [comment]
Thank for you review @paulbatum 
I tried to refacto MobileServiceClient functions to avoid duplication. I hope it will be correct.

Best regards,

File: sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Http/MobileServiceHttpClient.cs [hunk 1]
```diff
@@ -197,6 +198,7 @@ public MobileServiceHttpClient(IEnumerable<HttpMessageHandler> handlers, Uri app
         /// <param name="features">
         /// Value indicating which features of the SDK are being used in this call. Useful for telemetry.
         /// </param>
+		/// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> token to observe</param>
```
Comment (phvannor) [inline sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Http/MobileServiceHttpClient.cs]: nit: can you convert your tabs to spaces so its all in the same style.

**Comment by scastria** [comment]
What about all of the other Async methods in the Azure Mobile Client, like ToListAsync(), etc.?  Those need cancellation tokens too.

**Pull request merged**